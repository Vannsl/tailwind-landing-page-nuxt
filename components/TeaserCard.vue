<template>
  <div class="teaser-card">
    <div class="teaser-content">
      <a href="#">
        <p class="overline">
          <slot name="overline" />
        </p>
        <div class="headline">
          <slot name="headline" />
        </div>
        <p class="content">
          <slot name="content" />
        </p>
      </a>
    </div>
    <div v-if="!!action" class="teaser-action">
      <div>
        <button class="gradient" @click.prevent="onClick">{{ action }}</button>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'TeaserCard',
  props: {
    action: {
      type: String,
      default: ''
    }
  },
  methods: {
    onClick() {
      // NOTE: is fired, but not handled in this template
      this.$emit('clicked')
    }
  }
}
</script>

<style scoped>
.teaser-card {
  @apply w-full p-6 flex flex-col flex-grow flex-shrink;
}

@screen md {
  .teaser-card {
    @apply flex-1;
    /* todo generic width...?! */
  }
}

.teaser-content {
  @apply flex-1 bg-white rounded-t rounded-b-none overflow-hidden shadow;
}

.teaser-content > a {
  @apply flex flex-wrap no-underline;
}

.teaser-content > a:hover {
  @apply no-underline;
}

.overline {
  @apply w-full text-gray-600 text-xs px-6;
}

@screen md {
  .overline {
    @text-sm;
  }
}

.headline {
  @apply w-full font-bold text-xl text-gray-800 px-6;
}

.content {
  @apply text-gray-800 text-base px-6 mb-5;
}

.teaser-action {
  @apply flex-none mt-auto bg-white rounded-b rounded-t-none overflow-hidden shadow p-6;
}

.teaser-action > div {
  @apply flex items-center justify-end;
}

.teaser-action button {
  @apply mx-auto text-white font-bold rounded-full my-6 py-4 px-8 shadow-lg;
}

.teaser-action button:hover {
  @apply underline;
}

@screen lg {
  .teaser-action button {
    @apply mx-0;
  }
}
</style>
