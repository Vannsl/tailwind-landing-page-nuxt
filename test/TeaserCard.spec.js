import { shallowMount } from '@vue/test-utils'
import TeaserCard from '@/components/TeaserCard'

const factory = () => {
  return shallowMount(TeaserCard, {
    propsData: {
      action: 'Action'
    }
  })
}

describe('TeaserCard', () => {
  it('is a Vue instance', () => {
    const wrapper = factory()
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders properly', () => {
    const wrapper = factory()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('has the property action', () => {
    const wrapper = factory()
    expect(wrapper.props('action')).toBe('Action')
  })

  describe('when the user clicks on the button', () => {
    it('emits the event', () => {
      const wrapper = factory()
      wrapper.find('button').trigger('click')
      expect(wrapper.emitted().clicked).toBeTruthy()
    })
  })

  describe('when the property action is not an empty string', () => {
    it('renders the button', () => {
      const wrapper = factory()
      expect(wrapper.find('button').exists()).toBe(true)
    })
  })

  describe('when the property action is an empty string', () => {
    it('renders the button', () => {
      const wrapper = factory()
      wrapper.setProps({ action: '' })
      expect(wrapper.find('button').exists()).toBe(false)
    })
  })
})
