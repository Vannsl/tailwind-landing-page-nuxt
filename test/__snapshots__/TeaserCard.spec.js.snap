// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeaserCard renders properly 1`] = `"<div class=\\"teaser-card\\"><div class=\\"teaser-content\\"><a href=\\"#\\"><p class=\\"overline\\"></p> <div class=\\"headline\\"></div> <p class=\\"content\\"></p></a></div> <div class=\\"teaser-action\\"><div><button class=\\"gradient\\">Action</button></div></div></div>"`;
